resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DbReader.scala

```diff
@@ -36,6 +35,37 @@ object DbReader {
     sql"""select googleProject, initBucket from CLUSTER WHERE status="Deleted";"""
       .query[InitBucketToRemove]
 
+  // Return all non-deleted clusters with non-default nodepools that have apps that were all deleted
+  // or errored outside the grace period (1 hour)
+  //
+  // We are excluding clusters with only default nodepools running on them so we do not remove batch-pre-created clusters.
+  // We are calculating the grace period for cluster deletion assuming that the following are valid proxies for an app's last activity:
+  //    1. destroyedDate for deleted apps
+  //    2. dateAccessed for error'ed apps
+  // TODO: Read the grace period (hardcoded to '1 HOUR' below) from config
+  val kubernetesClustersToDeleteQuery =
+    sql"""
+            SELECT kc.id, kc.googleProject
+            FROM KUBERNETES_CLUSTER kc
+            WHERE
+              kc.status != "DELETED" AND
+              NOT EXISTS (
+                SELECT *
+                FROM NODEPOOL np
+                LEFT JOIN APP a ON np.id = a.nodepoolId
+                WHERE
+                  kc.id = np.clusterId AND np.isDefault = 0 AND
+                  (
+                    (a.status != "DELETED" AND a.status != "ERROR") OR
+                    (a.status = "DELETED" AND a.destroyedDate > now() - INTERVAL 1 HOUR) OR
+                    (a.status = "ERROR" AND a.dateAccessed > now() - INTERVAL 1 HOUR) OR
+                    (a.id IS NULL)
```

Comment:
@rtitle  ^^^ seems to take care of the exclusion of clusters with batch-pre-created nodepools. Added a [unit test](https://github.com/broadinstitute/leonardo-cron-jobs/pull/15/files#diff-d4eec159732c276b41661daac0de09192e20d7d261664c25dbc61d77bf0bb058R216) for that scenario as well.